modules/smtp/modules.php

```diff
@@ -117,13 +117,62 @@ public function process() {
 /**
  * @subpackage smtp/handler
  */
-class Hm_Handler_load_smtp_is_imap_forward extends Hm_Handler_Module
+class Hm_Handler_load_smtp_is_imap_forward_as_attachment extends Hm_Handler_Module
 {
     public function process() {
         if (!$this->module_is_supported('imap')) {
             return;
         }
 
+        if (array_key_exists('forward_as_attachment', $this->request->get)) {
+            $path = explode('_', $this->request->get['list_path']);
+            $imap = Hm_IMAP_List::connect($path[1]);
+            if ($imap && $imap->select_mailbox(hex2bin($path[2]))) {
+                $msg_struct = $imap->get_message_structure($this->request->get['uid']);
+                list($part, $msg_text) = $imap->get_first_message_part($this->request->get['uid'], 'text', 'plain', $msg_struct);
+                $msg_header = $imap->get_message_headers($this->request->get['uid']);
+                if (!array_key_exists('From', $msg_header) || count($msg_header) == 0) {
+                    return;
+                }
+                $content = $imap->get_message_content($this->request->get['uid'], 0, false, false);
```

Comment:
We need to make sure we get the full email as stored by the remote imap server including all headers and message parts.

Reply:
I think it will also make sense to forward the plain text / html parts as an eml and forward the attached files to the original email as separate attachments. Not sure - @marclaporte what do you think?

Reply:
I just checked and in Kolab (Roundcube), forwarding bundles everything in one .eml. I think this makes sense. If folks want to pick and choose what to forward, they can use "Edit as new message"

Reply:
OK, let's keep the whole eml including attachments type of forward for now.